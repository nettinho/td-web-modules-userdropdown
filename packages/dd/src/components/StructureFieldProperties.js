import React from "react";
import PropTypes from "prop-types";
import { List } from "semantic-ui-react";
import { compose } from "redux";
import { connect } from "react-redux";
import { injectIntl } from "react-intl";

export const StructureFieldProperties = ({
  intl: { formatMessage },
  type,
  nullable,
  precision
}) => (
  <List>
    {type && <List.Item icon="font" content={type} />}
    {nullable !== undefined && (
      <List.Item
        icon="minus"
        content={
          nullable
            ? formatMessage({ id: "structure.field.optional" })
            : formatMessage({ id: "structure.field.required" })
        }
      />
    )}
    {precision && <List.Item icon="quote right" content={precision} />}
  </List>
);

StructureFieldProperties.propTypes = {
  intl: PropTypes.object,
  type: PropTypes.string,
  nullable: PropTypes.bool,
  precision: PropTypes.string
};

const mapStateToProps = ({ structureField }) => ({
  ...structureField
});

export default compose(
  injectIntl,
  connect(mapStateToProps)
)(StructureFieldProperties);
