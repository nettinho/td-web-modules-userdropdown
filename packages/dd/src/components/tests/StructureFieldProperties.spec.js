import React from "react";
import { shallowWithIntl } from "intl-stub";
import { StructureFieldProperties } from "../StructureFieldProperties";

describe("<StructureFieldProperties />", () => {
  it("matches the latest snapshot", () => {
    const props = {
      id: 1,
      name: "f1",
      description: "dd",
      type: "tp",
      nullable: true,
      precision: "22"
    };
    const wrapper = shallowWithIntl(<StructureFieldProperties {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
