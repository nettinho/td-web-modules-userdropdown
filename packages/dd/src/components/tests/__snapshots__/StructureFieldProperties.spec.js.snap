// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructureFieldProperties /> matches the latest snapshot 1`] = `
<List>
  <ListItem
    content="tp"
    icon="font"
  />
  <ListItem
    content="structure.field.optional"
    icon="minus"
  />
  <ListItem
    content="22"
    icon="quote right"
  />
</List>
`;
